src/hist/basehist.py

```diff
@@ -94,6 +94,13 @@ def __init__(
                 warnings.warn(msg)
                 storage = storage()
             super().__init__(*args, storage=storage, metadata=metadata)  # type: ignore
+
+            disallowed_names = ["weight", "sample", "threads"]
```

Comment:
```suggestion
            disallowed_names = {"weight", "sample", "threads"}
```

Ultra minor, but this is slightly more logically correct. It's technically probably a bit slower, just because sets take longer to construct than lists/tuples unless you are constructing them and testing containership in the same line. Actually checking it is faster, so if you did it multiple times, sets are a clear win. But we aren't talking about a significant time in any case.

Reply:
Alright! That makes sense. :)